client/concordclient/include/client/concordclient/remote_update_recipient.hpp

```diff
@@ -46,6 +46,11 @@ struct Update {
   std::string span_context;
 };
 
-typedef std::variant<Update, EventGroup> EventVariant;
+struct SnapshotKVPair {
+  std::string key;
+  std::string val;
+};
+
+typedef std::variant<Update, EventGroup, SnapshotKVPair> RemoteRecepient;
```

Comment:
I am thinking that this variant contains the types of data we transfer. Maybe we can rename it to `ClientData` or similar. What do you think?

Reply:
There is an Inter Thread Communication (ITC) container. The name of that container is BasicUpdateQueue. That container, unlike vector, is not defined for all kinds of types. Instead, it's defined for a subset of types. This type represents those data types that the ITC can contain. The ITC contains a push method that blocks until the thread on the other end pops the data. So it's like a pipe abstraction. Now assume that there are 2 threads, the producer thread produces the data for the thread of the other end and the consumer will consume the data created in the other end. So this data acts as a recipient of remote data produced/consumed at a remote end. So it's named as RemoteRecepient, but if this name is not apt, then a better name should be RemoteData (if we want to view it as Data). But we should not attach it to the client as it's related to an ITC.

Reply:
I understand, thanks for the clarification. What I am suggesting is that we are sending some data in the variant and the value is not the recipient itself - instead, it is the data.

Reply:
Thanks, @dartdart26 I got it, let me rename this as RemoteData.